_inc/lib/class-jetpack-mapbox-helper.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Mapbox API helper.
+ *
+ * @package jetpack
+ */
+
+/**
+ * Class Jetpack_Mapbox_Helper
+ */
+class Jetpack_Mapbox_Helper {
+	/**
+	 * Site option key for the Mapbox service.
+	 *
+	 * @var string
+	 */
+	private static $site_option_key = 'mapbox_api_key';
+
+	/**
+	 * Transient key for the WordPress.com Mapbox access token.
+	 *
+	 * @var string
+	 */
+	private static $transient_key = 'wpcom_mapbox_access_token';
+
+	/**
+	 * Get the site's own Mapbox access token if set, or the WordPress.com's one otherwise.
+	 *
+	 * @return array An array containing the key (if any) and its source ("site" or "wpcom").
+	 */
+	public static function get_access_token() {
+		// If the site provides its own Mapbox access token, return it.
+		$service_api_key = Jetpack_Options::get_option( self::$site_option_key );
+		if ( $service_api_key ) {
+			return self::format_access_token( $service_api_key );
+		}
+
+		$site_id = self::get_wpcom_site_id();
+
+		// If on WordPress.com, try to return the access token straight away.
+		if ( self::is_wpcom() && defined( 'WPCOM_MAPBOX_ACCESS_TOKEN' ) ) {
+			jetpack_require_lib( 'mapbox-blacklist' );
+			return wpcom_is_site_blacklisted_from_map_block( $site_id )
+				? self::format_access_token()
+				: self::format_access_token( WPCOM_MAPBOX_ACCESS_TOKEN, 'wpcom' );
+		}
+
+		// If not on WordPress.com, return an empty access token.
```

Comment:
It may be worth precising the case of Atomic sites in this comment, to clarify things a bit.

Reply:
You're right, we even talked about this already! 🙂